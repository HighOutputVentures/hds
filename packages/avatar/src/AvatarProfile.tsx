import { SystemStyleObject } from "@chakra-ui/react";
import * as React from "react";
import Avatar, { AvatarBaseProps } from "./Avatar";
import { useActualSize } from "./hooks";
import { Clickable, ResponsiveSize } from "./types";

type AvatarProfileSize = "sm" | "md" | "lg";

type AvatarProfileBaseProps = {
  size?: AvatarProfileSize | ResponsiveSize<AvatarProfileSize>;
};

type InheritedAvatarProps = Pick<AvatarBaseProps, "src" | "name" | "verified">;

// prettier-ignore
export type AvatarProfileProps =
  SystemStyleObject &
  InheritedAvatarProps &
  AvatarProfileBaseProps &
  Clickable;

export default function AvatarProfile(props: AvatarProfileProps) {
  const { size, ...others } = Object.assign({ size: "md" }, props);

  const actualSize = useActualSize(size);
  const psuedoSize = { sm: "3xl", md: "4xl", lg: "5xl" } as const;

  return <Avatar size={psuedoSize[actualSize]} {...others} __elevated __bordered />;
}
