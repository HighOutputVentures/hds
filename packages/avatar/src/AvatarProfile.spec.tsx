// No need to do lots of tests here
// This just calls avatar underneath
// thus checking if avatar is rendered would be enough

import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import * as React from 'react';
import AvatarProfile from './AvatarProfile';

jest.mock('@chakra-ui/react', function mockUseBreakpointHook() {
  return {
    ...jest.requireActual('@chakra-ui/react'),
    useBreakpoint() {
      return 'lg';
    },
  };
});

describe('AvatarProfile', () => {
  it("Should render just like regular 'Avatar'", () => {
    const { queryByRole } = render(<AvatarProfile src="" name="John Doe" />);

    expect(queryByRole('img', { name: 'avatar' })).toBeDefined();
  });
});
