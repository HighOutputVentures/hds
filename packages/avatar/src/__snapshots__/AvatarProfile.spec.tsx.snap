// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarProfile Snapshot Should match snapshot 1`] = `
<span
  className="chakra-avatar css-1ruhx4w"
  data-bordered={true}
  data-clickable={false}
  data-elevated={true}
  data-fallback={true}
  data-online={false}
>
  <div
    aria-label="John Doe"
    className="chakra-avatar__initials css-0"
    role="img"
  >
    JD
  </div>
</span>
`;
